src/v2/normalized_packages.py

```diff
@@ -53,3 +57,65 @@ def dependency_graph(self) -> Dict[Package, Set[Package]]:
     def ecosystem(self):
         """Ecosystem value."""
         return self._ecosystem
+
+
+class GoNormalizedPackages(NormalizedPackages):
+    """Duplicate free list of GoNormalised Packages."""
+
+    def __init__(self, packages: List[Package], ecosystem: Ecosystem):
+        """Create NormalizedPackages by removing all duplicates from packages."""
+        self._packages = packages
+        self._ecosystem = ecosystem
+        self._dependency_graph: Dict[Package, Set[Package]] = defaultdict(set)
+        self._modules = []
+        self._version_map = {}
+        self.gh = GithubUtils()
+        self.pseudo = set()
+        for package in packages:
+            # clone without dependencies field
+            package.name, package.version, \
+                go_package_module = self.get_golang_metadata(package)
+            package_clone = Package(name=package.name, version=package.version)
+            if self.gh.is_pseudo_version(package.version):
+                self._modules.append(go_package_module)
+                self._version_map[package.name] = package.version
+                self.pseudo.add(package_clone)
```

Comment:
Lots of code duplicate w.r.t base class. The idea of a base class is to reuse as much implementation possible., the derived class should only handle the specialisation.

Reply:
I agree Arun, But For Golang, we need a splitting on  `package_name` and package_module` and hence generating a dependency tree after that only. I was avoiding calls to `super()` bcz I feel the overhead is considerable

Reply:
Ack!